frontend/src/middleware/page-views.ts

```diff
@@ -0,0 +1,36 @@
+import { useAnalytics } from "~/composables/use-analytics"
+
+import type { Middleware } from "@nuxt/types"
+
+/**
+ * This middleware sends events to Plausible.
+ *
+ * - For the initial page load, executed on the server, it sends a
+ *   `SERVER_RENDERED` event.
+ * - For subsequent page loads, executed on the client, it sends a
+ *   `VIEW_PAGE` event between every navigation.
+ *
+ * These events are in addition to the `pageview` events automatically sent by
+ * the Plausible integration.
```

Comment:
- The `SERVER_RENDERED` event was experimental (given the draft nature of this PR) to see how we could potentially send events from the server. This event is actually not needed because SSR or not, the pageview event is sent from the client for each page visited.

- The `VIEW_PAGE` event written in the RFC prior to choosing Plausible an a platform. If we were to go with a different platform (or build one ourselves), we'd have to add this `VIEW_PAGE` event but with Plausible mandating `pageview`, it's not needed.

Reply:
I intended to remove both of them before undrafting. The point of this PR being a draft was to run as many experiments as possible while other PRs are blocking it.